From 5ec009d5551c7736c75d43ae20bec1a055bb0c5a Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 13:18:21 -0400
Subject: [ethernet] r8169: initialize rtl8169_stats seqlock

Message-id: <1410182312-19560-3-git-send-email-ivecera@redhat.com>
Patchwork-id: 91989
O-Subject: [RHEL7.1 PATCH 02/13] r8169: initialize rtl8169_stats seqlock
Bugzilla: 1139247
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

BZ: #1139247

Upstream commit(s):
commit 340fea3d7f6a2657ddd0b48413cd81e8513357ed
Author: Kyle McMartin <kyle@redhat.com>
Date:   Mon Feb 24 20:12:28 2014 -0500

    r8169: initialize rtl8169_stats seqlock

    Boris reports he's seeing:
    > [    9.195943] INFO: trying to register non-static key.
    > [    9.196031] the code is fine but needs lockdep annotation.
    > [    9.196031] turning off the locking correctness validator.
    > [    9.196031] CPU: 1 PID: 933 Comm: modprobe Not tainted 3.14.0-rc4+ #1
    with the r8169 driver.

    These are occuring because the seqcount embedded in u64_stats_sync on
    32-bit SMP is uninitialized which is making lockdep unhappy.

    Signed-off-by: Kyle McMartin <kyle@redhat.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/realtek/r8169.c b/drivers/net/ethernet/realtek/r8169.c
index 91a67ae..e977965 100644
--- a/drivers/net/ethernet/realtek/r8169.c
+++ b/drivers/net/ethernet/realtek/r8169.c
@@ -7118,6 +7118,8 @@ rtl_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
  }
 
  mutex_init(&tp->wk.mutex);
+ u64_stats_init(&tp->rx_stats.syncp);
+ u64_stats_init(&tp->tx_stats.syncp);
 
  /* Get MAC address */
  for (i = 0; i < ETH_ALEN; i++)
-- 
1.7.1